### DISCUSSION TITLE
Can Quarto generate self-contained websites or books?

### DISCUSSION THREAD

**Discussion by ezherman** [quarto-dev/quarto-cli#10301] [Q&A] [answered]
### Description

I am looking to generate a self-contained Quarto website or book which will bundle multiple reports into one HTML file. The `embed-resources` option works great at the report level, however I have not found a way to enable this at the website or book level. Is this at all possible? I realise this question has been asked before - I was hoping to learn whether this feature has since been implemented or whether it is planned for an upcoming release. Thanks!

**Comment by mcanouil** [comment, marked as answer]
That's very unlikely HTML Website/Book will ever work with `embed-resources`.
Also, even if it is implemented it would to copy absolutely all CSS/JavaScript/etc. in every single pages leading to very big websites leading to longer loading times, etc.

Note that embed-resources has nothing to do with your definition of "self-contained" which is "single file document" from what you wrote.
HTML Website/Book project will never be "self-contained" as per your definition. That's not something possible.
